dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageErrorWithReason.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.classpath;
+
+import java.net.URL;
+import javax.annotation.Nullable;
+
+/**
+ * Interface to provide common fields for different types of static linkage errors.
+ */
+interface LinkageErrorWithReason {
+
+  /**
+   * Returns the location of the target class in a symbol reference; null if the target class is not
+   * found in the class path or the source location is unavailable.
+   */
+  @Nullable
+  URL getTargetClassLocation();
+
+  /**
+   * Returns the reason why a symbol reference is marked as a linkage error.
+   */
+  Reason getReason();
+
+  /**
+   * Reason to distinguish the cause of a static linkage error against a symbol reference.
+   */
+  enum Reason {
+    /**
+     * The target class of the symbol reference is not found in the class path.
+     */
+    TARGET_CLASS_NOT_FOUND,
```

Comment:
The reasons all apply to the target symbol, right? We don't have to specify that it's the _target_ class that's not found. How about just `CLASS_NOT_FOUND`?